APCAppCore/APCAppCore/Library/Categories/NewDataCollector/APCPassiveDisplacementTrackingDataUploader.m

```diff
@@ -0,0 +1,293 @@
+    if (manager.location.floor == nil)
+    {
+        floor = NSLocalizedString(@"not available", nil);
+    }
+    else
+    {
+        NSInteger level = manager.location.floor.level;
+        floor = [NSString stringWithFormat:@"%ld", (long)level];
+    }
+    
+    //  Altitude of the location can be positive (above sea level) or negative (below sea level).
+    double      altitude                = manager.location.altitude;
+    NSString*   altitudeUnit            = @"meters";
+    NSString*   horizontalAccuracy      = [NSString stringWithFormat:@"%f", manager.location.horizontalAccuracy];
+    NSString*   horizontalAccuracyUnit  = @"meters";
+    NSString*   verticalAccuracy        = [NSString stringWithFormat:@"%f", manager.location.verticalAccuracy];
+    NSString*   verticalAccuracyUnit    = @"meters";
+    
+    return  @[timestamp, distance, unit, @(direction), directionUnit, speed, speedUnit, floor, @(altitude), altitudeUnit, horizontalAccuracy, horizontalAccuracyUnit, verticalAccuracy, verticalAccuracyUnit];
+}
+
+
+- (void)didRecieveUpdateWithLocationManager:(CLLocationManager*)manager withUpdateLocations:(NSArray*)locations
+{
+    __weak typeof(self) weakSelf = self;
+    
+    [self.healthKitCollectorQueue addOperationWithBlock:^
+    {
+        __typeof(self)  strongSelf  = weakSelf;
+        NSArray*        result      = nil;
+        
+        if ((self.baseTrackingLocation.coordinate.latitude == 0.0) && (self.baseTrackingLocation.coordinate.longitude == 0.0))
+        {
+            self.baseTrackingLocation = [locations firstObject];
+            self.mostRecentUpdatedLocation = self.baseTrackingLocation;
+            
+            if ([locations count] >= 1)
+            {
+                CLLocationDistance  distanceFromReferencePoint = [self.baseTrackingLocation distanceFromLocation:manager.location];
+                
+                result = [self locationDictionaryWithLocationManager:manager
+                                          distanceFromReferencePoint:distanceFromReferencePoint
+                                                 andPreviousLocation:self.baseTrackingLocation];
+            }
+        }
+        else
+        {
+            self.baseTrackingLocation = self.mostRecentUpdatedLocation;
+            self.mostRecentUpdatedLocation = manager.location;
+            
+            CLLocationDistance  distanceFromReferencePoint = [self.baseTrackingLocation distanceFromLocation:manager.location];
+            result = [self locationDictionaryWithLocationManager:manager
+                                      distanceFromReferencePoint:distanceFromReferencePoint
+                                             andPreviousLocation:self.baseTrackingLocation];
+        }
+        
+        //Send to delegate
+        if (result)
+        {  
+            NSString *stringToWrite = [NSString stringWithFormat:@"%@,%@,%@,%@,%@,%@,%@,%@,%@,%@,%@,%@,%@,%@\n",
```

Comment:
I specifically didn't say that because you'll also need the new-line and it makes no sense to copy the entire string just to do that.